src/components/dialog/dialog.js

```diff
@@ -431,7 +437,11 @@ function MdDialogProvider($$interimElementProvider) {
         '<md-dialog md-theme="{{ dialog.theme }}" aria-label="{{ dialog.ariaLabel }}" ng-class="dialog.css">',
         '  <md-dialog-content class="md-dialog-content" role="document" tabIndex="-1">',
         '    <h2 class="md-title">{{ dialog.title }}</h2>',
-        '    <div class="md-dialog-content-body" md-template="::dialog.mdContent"></div>',
+        '    <div ng-if="dialog.mdHtmlContent" class="md-dialog-content-body" ',
+        '        ng-bind-html="dialog.mdHtmlContent"></div>',
+        '    <div ng-if="!dialog.mdHtmlContent" class="md-dialog-content-body">',
+        '      {{dialog.mdTextContent}}',
+        '    </div>',
```

Comment:
Not a big deal (since usually there is max one dialog at any given moment), but using `ngSwitch` and one-time-bindings should be (unnoticably) better for performance.